Bottom: 943bd81fe8922d43c7b6e95c9e9d86581b5b54e8
Top:    f3aa2ffc22b0870255176d871bcfa55ddeda49d5
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-05-01 23:29:23 +0100

hid/gtk: Fix conversion from world to screen in ghid_invalidate_lr

I accidentally muddled the Vx and Vy conversions routines for the right
and top coordinates. In general, this resulted in the incorrect region
being invalidated - manifesting as artaefacts on the the screen, such as
when dragging objects around.

Also drop the (double) cast.. having checked, Vx and Vy take integer
arguments. I'm not sure why the rest of this file casts to double.


---

diff --git a/src/hid/gtk/gtkhid-gdk.c b/src/hid/gtk/gtkhid-gdk.c
index 0b0b157..4534905 100644
--- a/src/hid/gtk/gtkhid-gdk.c
+++ b/src/hid/gtk/gtkhid-gdk.c
@@ -800,10 +800,10 @@ ghid_invalidate_lr (int left, int right, int top, int bottom)
   int minx, maxx, miny, maxy;
   GdkRectangle rect;
 
-  dleft = Vx ((double) left);
-  dright = Vy ((double) right);
-  dtop = Vx ((double) top);
-  dbottom = Vy ((double) bottom);
+  dleft = Vx (left);
+  dright = Vx (right);
+  dtop = Vy (top);
+  dbottom = Vy (bottom);
 
   minx = MIN (dleft, dright);
   maxx = MAX (dleft, dright);
